app/code/Magento/Catalog/Model/Product/Type.php

```diff
@@ -116,8 +113,8 @@ public function __construct(
     /**
      * Factory to product singleton product type instances
      *
-     * @param   \Magento\Catalog\Model\Product $product
-     * @return  \Magento\Catalog\Model\Product\Type\AbstractType
+     * @param ProductInterface $product
```

Comment:
@slavvka is this type also might be reverted ? or we can leave `\Magento\Framework\Pricing\PriceInfoInterface `?

Reply:
Use full qualified type for ProductInterface. Don't import it.

Reply:
ok, i mean the in this case return type was changed from  `\Magento\Catalog\Model\Product `to `Magento\Catalog\Api\Data\ProductInterface`

Reply:
Exactly, without it there's no reason for making this PR

Reply:
done :heavy_check_mark: